spyder/plugins/maininterpreter.py

```diff
@@ -153,6 +158,28 @@ def python_executable_switched(self, custom):
             if custom:
                 self.warn_python_compatibility(cust_pyexec)
 
+    def is_python_interpreter(self, pyexec):
+        if not osp.isfile(pyexec):
+            return False
+        args = ["-c", "print('valid_python_interpreter')"]

```

Comment:
The problem with point 1. on Linux is that `python{2,3}` are symlinks, so you should follow them to the file they point to and check that it's a binary.

Also, about 1., users could introduce `python2.7` or `python3.5`, so that should be covered too here.